Argo/Functions/flatReduce.swift

```diff
@@ -17,7 +17,7 @@
   - returns: The result of iterating the combinator over every element of the
              sequence and flattening the result
 */
-public func flatReduce<S: SequenceType, U>(sequence: S, initial: U, @noescape combine: (U, S.Generator.Element) -> Decoded<U>) -> Decoded<U> {
+public func flatReduce<S: Sequence, U>(sequence: S, initial: U, combine: @noescape (U, S.Iterator.Element) -> Decoded<U>) -> Decoded<U> {
```

Comment:
I think we should maintain the existing API here. I'd vote for ignoring the external label on `sequence`.